src/integrationtests/java/com/aws/greengrass/integrationtests/certificateauthority/OfflineAuthenticationTest.java

```diff
@@ -362,4 +379,41 @@ void GIVEN_clientConnectsWhileOnline_WHEN_offlineAndCertDetachedFromThing_THEN_b
         network.goOnline();
         assertThrows(AuthenticationException.class, () -> {connectToCore(thingOne.get(), clientPem);});
     }
+
+    @Test
+    void GIVEN_clientConnectsWhileOnline_WHEN_storedPemIsCorrupted_THEN_clientCanStillConnectOnlineOrOffline(
+            ExtensionContext context) throws Exception {
+        ignoreExceptionOfType(context, NoSuchFileException.class);
+        ignoreExceptionOfType(context, InvalidCertificateException.class);
+        // Given
+        network.goOnline();
+
+        // Configure the things attached to the core
+        List<X509Certificate> clientCertificates = createClientCertificates(1);
+        String clientPem = CertificateHelper.toPem(clientCertificates.get(0));
+        Certificate certificate = Certificate.fromPem(clientPem);
+        Supplier<String> thingOne =  () -> "ThingOne";
+        iotAuthClientFake.attachCertificateToThing(thingOne.get(), clientPem);
+        iotAuthClientFake.attachThingToCore(thingOne);
+        iotAuthClientFake.activateCert(clientPem);
+
+        runNucleusWithConfig("config.yaml");
+        connectToCore(thingOne.get(), clientPem);
+
+        // When
+        corruptStoredClientCertificate(clientPem);
+
+        // Then
+
+        // Assert cert pem is corrupted
+        ClientCertificateStore pemStore = kernel.getContext().get(ClientCertificateStore.class);
+        String storeCertificatePem = pemStore.getPem(certificate.getCertificateId()).get();
+        assertThrows(InvalidCertificateException.class, () -> Certificate.fromPem(storeCertificatePem));
+
+        // Assert that authenticating offline or online is not affected
+        network.goOffline();
+        assertNotNull(connectToCore(thingOne.get(), clientPem));
+        network.goOnline();
+        assertNotNull(connectToCore(thingOne.get(), clientPem));
```

Comment:
Do we fix the pem on disk by this point? How do we eventually recover from the bad storage? If we do recover by this point, let's verify that the pem on disk is what we want.

Reply:
Uhh that is a good one. It should get fixed. Let me add it 🧠

Reply:
I agree this is something we should have, but at this point I'm inclined to cut a backlog item to avoid scope creep. We're frozen at this point